### PR TITLE
Make some navigation classes and methods public for extended usability

### PR DESCRIPTION
**Pull request by BackPackerDz**
This PR updates the access modifiers of `ViewRouteListener` and `NavigationViewRouter`, along with three of its methods (`findRouteFrom`, `updateLocation`), to public. These changes are essential to allow developers to handle rerouting more directly within their applications. Without these modifications, developers are forced to rely on less desirable solutions such as duplicating internal code or employing fragile workarounds. Making these classes and methods public simplifies the development process and enhances the flexibility of the navigation functionality.

I welcome feedback on these changes and am open to discussing alternative approaches. Please share any suggestions or insights you have.

Several useful classes in the library are internal, limiting their direct usability. I’d appreciate insights into the decision for such accessibility settings to align our development efforts better.

### CODE REVIEW

File: .github/workflows/publish_library.yaml
Comment (boldtrn) [inline .github/workflows/publish_library.yaml:1]: What is this change about? Where does this release to?

**Review by boldtrn — CHANGES_REQUESTED** [review]
I don't mind changing some methods or classes to public to help with overwriting certain features. I am not sure about the publishing script. @Fabi755 do you have an any other input on this?

File: .github/workflows/publish_library.yaml
Comment (BackPackerDz) [reply .github/workflows/publish_library.yaml:1]: My apologies, I didn't notice that pushing to our own fork also updated this branch.

**Review by boldtrn — APPROVED** [review]
LGTM 👍

**Pull request merged**